semver.py

```diff
@@ -168,6 +168,12 @@ def __iter__(self):
         for v in self._astuple():
             yield v
 
+    def __getitem__(self, index):
+        """Implement getitem. This automatically strips empty parts of the
+           version from the iterable from which the index is taken."""
```

Comment:
Could you change:

```suggestion
           version from the iterable from which the index is taken.

           :param int index: a positive or negative integer
           :raises: IndexError, if index is beyond the range
           """
```

Reply:
@tomschr according to current implementation, index can be negative.

@tlaferriere I think there is a subtle bug in your implementation:
```
    def test_example():
        v = VersionInfo.parse('1.2.3+build0')
>       assert v[:] == (1, 2, 3, None, 'build0')
E       AssertionError: assert (1, 2, 3, 'build0') == (1, 2, 3, None, 'build0')
E         At index 3 diff: 'build0' != None
E         Right contains more items, first extra item: 'build0'
E         Use -v to get the full diff
```
i.e. if "prerelease" part is missing and "metadata" is present, after using __getitem__ , "metadata" appears at index for "prerelease"

Reply:
@ppkt 
> according to current implementation, index can be negative.

You are right, I've corrected the GH suggestion. Thanks! :+1:

Reply:
After much deliberation with myself and an increasingly complex function, I have decided to rewrite it (multiple times). I believe I have made this implementation as simple as possible. Turns out the subtle bug mentioned by @ppkt was a symptom of a few bad design decisions on my part. Now it doesn't support negative indices.